coalib/results/TextRange.py

```diff
@@ -89,7 +89,20 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        return (self.is_behind(self.start, other.end) and
+                self.is_behind(other.start, self.end))
+
+    @staticmethod
+    def is_behind(first_position, second_position):
```

Comment:
what about actually overriding the `<` and `>` operators for `TextPosition` with your code? Then you can just do inside `overlaps`, then you can keep the old method?